gems/aws-sdk-s3/lib/aws-sdk-s3/multipart_stream_uploader.rb

```diff
@@ -62,12 +62,18 @@ def upload_parts(upload_id, options, &block)
         completed = Queue.new
         errors = IO.pipe do |read_pipe, write_pipe|
           threads = upload_in_threads(read_pipe, completed, upload_part_opts(options).merge(upload_id: upload_id))
-          block.call(write_pipe)
-          write_pipe.close
+          begin
+            block.call(write_pipe)
+          ensure
+            write_pipe.close # Ensure the pipe is closed to avoid https://github.com/jruby/jruby/issues/6111
```

Comment:
Minor: please put the comment on the prior line

Reply:
Sure :)